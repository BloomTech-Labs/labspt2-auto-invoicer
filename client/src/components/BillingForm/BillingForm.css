.billing_form {
  align-items: center;
  display: flex;
  justify-content: space-between;
  width: 600px;
}

.payment_area {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding: 0 5px;
  width: 300px;
}

.payment_area button {
  border: 1px inset gray;
  border-radius: 3px;
  line-height: 2.5;
  font-size: 2rem;
  padding: 5px 0;
  margin: 50px 0 0 0;
  width: 100%;
  box-shadow: 3px 3px 3px lightgray;
}

.payment_area button:hover {
  border: 1px outset gray;
  background: lightgray;
  cursor: pointer;
}
