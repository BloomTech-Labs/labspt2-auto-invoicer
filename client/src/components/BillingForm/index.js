import React from "react";

// import components here
import PaymentInfo from "../PaymentInfo";
import PaymentRadio from "../PaymentRadio";
import Button from "../reusableComponents/Button";
// import css here
import "./BillingForm.css";

class BillingForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      cc: "",
      exp: "",
      cvv: "",
      title: "Payment Info"
    };
  }
  changeHandler = e => {
    return this.setState({ [e.target.name]: e.target.value });
  };
  createPaymentObject = e => {
    e.preventDefault();

    console.log(this.state);
  };
  render() {
    const paymentInfo = Object.create(this.state);
    return (
      <form className="billing_form" onSubmit={this.createPaymentObject}>
        <div className="payment_area">
          <PaymentInfo
            paymentInfo={paymentInfo}
            changeHandler={this.changeHandler}
          />
          <Button>Buy Now</Button>
        </div>
        <section>
          <PaymentRadio label="9.99">
            <p>1 Month Subscription - $9.99</p>
          </PaymentRadio>
          <PaymentRadio label="0.99">
            <p>1 Invoice - $0.99</p>
          </PaymentRadio>
        </section>
      </form>
    );
  }
}

export default BillingForm;
