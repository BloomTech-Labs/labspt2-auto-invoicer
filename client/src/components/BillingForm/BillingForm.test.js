import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import BillingForm from "../BillingForm";

describe("<BillingForm />", () => {
  it("renders Billing Form", () => {
    const { asFragment } = render(<BillingForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
