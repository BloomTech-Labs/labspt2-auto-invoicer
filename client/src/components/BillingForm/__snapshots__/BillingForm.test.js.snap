// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillingForm /> renders Billing Form 1`] = `
<DocumentFragment>
  <form
    class="billing_form"
  >
    <div
      class="payment_area"
    >
      <fieldset
        class="payment_info"
      >
        <legend>
          Payment Info
        </legend>
        <input
          maxlength="16"
          name="cc"
          pattern="[0-9]{16}"
          placeholder="Credit Card #"
          required=""
          type="text"
          value=""
        />
        <input
          maxlength="4"
          name="exp"
          pattern="[0-9]{4}"
          placeholder="Expiration Date"
          required=""
          type="text"
          value=""
        />
        <input
          maxlength="3"
          name="cvv"
          pattern="[0-9]{3}"
          placeholder="Cvv"
          required=""
          type="text"
          value=""
        />
      </fieldset>
      <button
        class="button"
        type="submit"
      >
        Buy Now
      </button>
    </div>
    <section>
      <label
        class="payment_checkboxes"
        for="9.99"
      >
        <input
          id="9.99"
          name="payment"
          type="radio"
          value="9.99"
        />
        <p>
          1 Month Subscription - $9.99
        </p>
      </label>
      <label
        class="payment_checkboxes"
        for="0.99"
      >
        <input
          id="0.99"
          name="payment"
          type="radio"
          value="0.99"
        />
        <p>
          1 Invoice - $0.99
        </p>
      </label>
    </section>
  </form>
</DocumentFragment>
`;
